### PR TITLE
feat(rds/instance): rds instance creation supports case-sensitive table names

### PR DESCRIPTION
**Pull request by wanglin100**
**What this PR does / why we need it**:

rds instance creation supports case-sensitive table names

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

**Review by iskey — APPROVED** [review]
/lgtm

File: docs/resources/rds_instance.md [hunk 1]
```diff
@@ -200,6 +200,14 @@ The following arguments are supported:
   -> **NOTE:** async indicates the asynchronous replication mode. semisync indicates the semi-synchronous replication
   mode. sync indicates the synchronous replication mode.
 
+* `lower_case_table_names` - (Optional, String, ForceNew) Specifies the case-sensitive state of the database table name,
+  the default value is "1". Changing this parameter will create a new resource.
+    + 0: Table names are stored as fixed and table names are case-sensitive.
+    + 1: Table names will be stored in lower case and table names are not case-sensitive.
+
+  -> **NOTE:** When restoring an existing instance, the case setting of the table name of the target instance is
+  inconsistent with that of the original instance, which may cause the restoration to fail.
```
Comment (ShiChangkuo) [inline docs/resources/rds_instance.md]: we can delete this note as user can not restore the instance with terraform.

File: docs/resources/rds_instance.md [hunk 2]
```diff
@@ -200,6 +200,14 @@ The following arguments are supported:
   -> **NOTE:** async indicates the asynchronous replication mode. semisync indicates the semi-synchronous replication
   mode. sync indicates the synchronous replication mode.
 
+* `lower_case_table_names` - (Optional, String, ForceNew) Specifies the case-sensitive state of the database table name,
```
Comment (ShiChangkuo) [inline docs/resources/rds_instance.md:203]: as the parameter can not be imported, please add a notification in Import section.

File: docs/resources/rds_instance.md (hunk 1 above)
Comment (wanglin100) [reply docs/resources/rds_instance.md]: ok, I have modified it

File: docs/resources/rds_instance.md (hunk 2 above)
Comment (wanglin100) [reply docs/resources/rds_instance.md:203]: Ok, I have modified it

**Pull request merged**